pkg/config/externalname.go

```diff
@@ -90,6 +93,18 @@ func TemplatedStringAsIdentifier(nameFieldPath, tmpl string) ExternalName {
 	if err != nil {
 		panic(errors.Wrap(err, "cannot parse template"))
 	}
+
+	// Note(turkenh): If a parameter is used in the external name template,
+	// it is an identifier field.
+	var identifierFields []string
+	for _, node := range t.Root.Nodes {
+		if node.Type() == parse.NodeAction {
+			match := parameterPattern.FindStringSubmatch(node.String())
```

Comment:
As far as I understand, we check the {{ .parameters.<field_name> }} for catching the root level required fields. This will cover some external name configurations if we use this convention. However, as you know, we also have another external name configurations, for example `IdentifierFromProvider`. For this case, we rely to the API schema of resource for the required fields while constructing the ID by terraform. What do you think about this type of cases?

Reply:
In the case of `IdentifierFromProvider`, provider assigns an id to the resource and we don't need (or at least know) parameters that will be included in that identifier. Noticed some helper methods in provider-aws like [FormattedIdentifierFromProvider](https://github.com/upbound/provider-aws/blob/93dc6725373000ef7af8803ee9e9ed94031eac71/config/externalname.go#L2737) which should be updated to set relevant parameters as identifier fields.

In upjet, noticed that we need to mark the parameter as identifier in `ParameterAsIdentifier` method and updated accordingly.

Reply:
Responded in the other thread: https://github.com/upbound/upjet/pull/166#discussion_r1162596296